metricbeat/scripts/mage/docs_collector.go

```diff
@@ -0,0 +1,512 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package mage
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+
+	"github.com/elastic/beats/dev-tools/mage"
+)
+
+// moduleData provides module-level data that will be used to populate the module list
+type moduleData struct {
+	Path       string
+	Base       string
+	Title      string `yaml:"title"`
+	Release    string `yaml:"release"`
+	Dashboards bool
+	Settings   []string `yaml:"settings"`
+	CfgFile    string
+	Asciidoc   string
+	Metricsets []metricsetData
+}
+
+// checkConfig checks to see if the config list has a particular item
+func (m moduleData) checkConfig(key string) bool {
+	for _, item := range m.Settings {
+		if item == key {
+			return true
+		}
+	}
+	return false
+}
+
+type metricsetData struct {
+	Path       string
+	Title      string
+	Link       string
+	Release    string
+	DataExists bool
+}
+
+var generatedNote = `////
+This file is generated! See scripts/docs_collector.py
+////
+
+`
+
+var moduleExampleConfig = `
+
+[float]
+=== Example configuration
+
+The %s module supports the standard configuration options that are described
+in <<configuration-metricbeat>>. Here is an example configuration:
+
+[source,yaml]
+----
+metricbeat.modules:
+`
+
+var metricsetFields = `
+
+==== Fields
+
+For a description of each field in the metricset, see the
+<<exported-fields-%s,exported fields>> section.
+
+`
+
+// getRelease gets the release tag. This kind of logic is needed because any
+// module/metricset missing a release is categorized as "experimental"
+// There are three valid tags: experimental, beta, ga
+func getRelease(rel string) (string, error) {
+	switch rel {
+	case "ga":
+		return "ga", nil
+	case "beta":
+		return "beta", nil
+	case "":
```

Comment:
`experimental` should still be an option :slightly_smiling_face: 
```suggestion
	case "experimental", "":
```

Reply:
oops!